ACEmulator/ACE
# Add server settings for quest xp and lum modifiers

**Purpose:**
- Allow server operator to be able to offer xp and luminance modifications explicitly for quests/emotes versus killing and other sources.
- Gives the server operator more flexibility but my personal motivation is for a server operator to encourage questing.

**Notes**
- Available via 2 server properties.  Will need to update the ACE Commands list if approved/merged.  https://github.com/ACEmulator/ACE/wiki/ACE-Commands#in-gameclient-commands
- The default is 1.0 for no change to existing server operators
- Does not undo any XP changes from the existing "luminance_modifier" and "xp_modifier" properties.  Those continue to work as before.
- These new properties are multiplicative with the old properties but just for quest/emote based XP.

**Example**
A server operator wants 2.0 general XP and 3.0 quest xp.  The operator would set "xp_modifier" to 2.0.  The operator would set "quest_xp_modifier" to 1.5.

**Testing**

- **_Testing complete.  Ready for Review_**
- Verify creature killing based xp is not affected by the change
- Verify skill xp is not affected by the change
- Verify item XP IS affected by the change - Player::GrantXP() handles this for kill and quest xp and is unchanged.
- Verify quest XP is multiplicative with "xp_modifier" and "quest_xp_modifier" for standard xp awards
- Same as last with non-shared xp
- Same as last with fractional level xp - includes real quests and things like colo ring xp
- Similar as above tests but with Lum.  The rules for lum are much simpler though.
- Limits testing
